nm: iterate the main context when refresh required

cathay4t opened the pull request:
In CI, we noticed nmstate might not able to retried applied config
for DHCP interface even with retry. This might caused by outdated cache
in `NM.Client` and cache is not refreshed after simple sleep between
verification.

The fix is iterate the main context till all event processed when
refreshing is issued.

ffmancera reviewed (changes requested):
The commit title is `WIP:..` could you change it?

cathay4t committed:
nm: iterate the main context when refresh required

In CI, we noticed nmstate might not able to retried applied config
for DHCP interface even with retry. This might caused by outdated cache
in `NM.Client` and cache is not refreshed after simple sleep between
verification.

The fix is iterate the main context till all event processed when
refreshing is issued.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
